airbyte-cdk/python/airbyte_cdk/base_python/cdk/streams/http.py

```diff
@@ -204,23 +204,23 @@ def _send_request(self, request: requests.PreparedRequest) -> requests.Response:
     def read_records(
         self,
         sync_mode: SyncMode,
-        stream_slice: Optional[Mapping[str, Any]] = None,
-        stream_state: Optional[Mapping[str, Any]] = None,
         cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
     ) -> Iterable[Mapping[str, Any]]:
         stream_state = stream_state or {}
         args = {"stream_state": stream_state, "stream_slice": stream_slice}
         pagination_complete = False
         while not pagination_complete:
             request = self._create_prepared_request(
-                path=self.path(**args),
-                headers=dict(self.request_headers(**args), **self.authenticator.get_auth_header()),
-                params=self.request_params(**args),
-                json=self.request_body_json(**args),
+                path=self.path(args),
```

Comment:
This seems incorrect. `**args` and `args` are different. The former denests keyword args from a dictionary, the latter passes a dict as an argument.

Reply:
Okay I figured this out - the header and request_body json functions do not accept an Optional with their first parameter, which is causing this error:
```
➜  python git:(davinchia/cdk-mypy) ✗ mypy airbyte_cdk/base_python
airbyte_cdk/base_python/cdk/streams/http.py:218: error: Argument 1 to "request_params" of "HttpStream" has incompatible type "**Dict[str, Optional[Mapping[str, Any]]]"; expected "Mapping[str, Any]"
airbyte_cdk/base_python/cdk/streams/http.py:219: error: Argument 1 to "request_body_json" of "HttpStream" has incompatible type "**Dict[str, Optional[Mapping[str, Any]]]"; expected "Mapping[str, Any]"
Found 2 errors in 1 file (checked 22 source files)
```

I think the function signature should continue being specific (instead of accepting Optional), so I'm going to break out the arguments. What do you think?